Deprecation warning when referencing amp task

In my `build.gradle`, I have a reference to the `amp` task for publishing as follows:

```
publishing {
    publications {
        mavenJava(MavenPublication) {
            from components.java
            artifact sourcesJar
            artifact javadocJar
            artifact amp
        }
    }
}
```

I noticed Gradle gives deprecation warnings for this:

```
> Configure project :alfred-telemetry-platform
SourceDirectorySetFactory has been deprecated. This is scheduled to be removed in Gradle 6.0. Please use the ObjectFactory service to create instances of SourceDirectorySet instead.
Using method ObjectFactory.property() method to create a property of type Map<K, V> has been deprecated. This will fail with an error in Gradle 6.0. Please use the ObjectFactory.mapProperty() method instead.
The ProjectLayout.fileProperty() method has been deprecated. This is scheduled to be removed in Gradle 6.0. Please use the ObjectFactory.fileProperty() method instead.
        at build_5roou882mjo74o10ivrlkjzkn$_run_closure9$_closure18$_closure19.doCall(/home/daan/ws/xenit/alfred-telemetry/alfred-telemetry-platform/build.gradle:86)
        (Run with --stacktrace to get the full stack trace of this deprecation warning.)

BUILD SUCCESSFUL in 8s
34 actionable tasks: 34 executed

```

Relevant part of the stacktrace:
```
> Configure project :alfred-telemetry-platform
SourceDirectorySetFactory has been deprecated. This is scheduled to be removed in Gradle 6.0. Please use the ObjectFactory service to create instances of SourceDirectorySet instead.
        at org.gradle.api.internal.file.DefaultSourceDirectorySetFactory.create(DefaultSourceDirectorySetFactory.java:31)
        at eu.xenit.gradle.alfrescosdk.internal.tasks.DefaultAmpSourceSetConfiguration.<init>(DefaultAmpSourceSetConfiguration.java:25)
        at eu.xenit.gradle.alfrescosdk.internal.tasks.DefaultAmpSourceSet.<init>(DefaultAmpSourceSet.java:23)
        at eu.xenit.gradle.alfrescosdk.AmpBasePlugin.lambda$configureSourceSetDefaults$1(AmpBasePlugin.java:64)
        at org.gradle.api.internal.DefaultCollectionCallbackActionDecorator$BuildOperationEmittingAction$1$1.run(DefaultCollectionCallbackActionDecorator.java:100)
```

As predicted: issue when using Gradle 6.
(See #33) 
```
* What went wrong:
An exception occurred applying plugin request [id: 'eu.xenit.amp', version: '1.0.0']
> Failed to apply plugin [class 'eu.xenit.gradle.alfrescosdk.AmpBasePlugin']
   > Please use the ObjectFactory.mapProperty() method to create a property of type Map<K, V>.

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.plugins.InvalidPluginException: An exception occurred applying plugin request [id: 'eu.xenit.amp', version: '1.0.0']
        at org.gradle.plugin.use.internal.DefaultPluginRequestApplicator.exceptionOccurred(DefaultPluginRequestApplicator.java:253)
        at org.gradle.plugin.use.internal.DefaultPluginRequestApplicator.applyPlugin(DefaultPluginRequestApplicator.java:235)
        at org.gradle.plugin.use.internal.DefaultPluginRequestApplicator.applyPlugins(DefaultPluginRequestApplicator.java:151)
        at org.gradle.configuration.DefaultScriptPluginFactory$ScriptPluginImpl.apply(DefaultScriptPluginFactory.java:216)
        at org.gradle.configuration.BuildOperationScriptPlugin$1$1.run(BuildOperationScriptPlugin.java:69)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.configuration.BuildOperationScriptPlugin$1.execute(BuildOperationScriptPlugin.java:66)
        at org.gradle.configuration.BuildOperationScriptPlugin$1.execute(BuildOperationScriptPlugin.java:63)
        at org.gradle.configuration.internal.DefaultUserCodeApplicationContext.apply(DefaultUserCodeApplicationContext.java:49)
        at org.gradle.configuration.BuildOperationScriptPlugin.apply(BuildOperationScriptPlugin.java:63)
        at org.gradle.configuration.project.BuildScriptProcessor$1.run(BuildScriptProcessor.java:45)
        at org.gradle.internal.Factories$1.create(Factories.java:26)
        at org.gradle.api.internal.project.DefaultProjectStateRegistry$ProjectStateImpl.withMutableState(DefaultProjectStateRegistry.java:212)
        at org.gradle.api.internal.project.DefaultProjectStateRegistry$ProjectStateImpl.withMutableState(DefaultProjectStateRegistry.java:193)
        at org.gradle.configuration.project.BuildScriptProcessor.execute(BuildScriptProcessor.java:42)
        at org.gradle.configuration.project.BuildScriptProcessor.execute(BuildScriptProcessor.java:26)
        at org.gradle.configuration.project.ConfigureActionsProjectEvaluator.evaluate(ConfigureActionsProjectEvaluator.java:35)
        at org.gradle.configuration.project.LifecycleProjectEvaluator$EvaluateProject$1.run(LifecycleProjectEvaluator.java:107)
        at org.gradle.internal.Factories$1.create(Factories.java:26)
        at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:189)
        at org.gradle.internal.work.StopShieldingWorkerLeaseService.withLocks(StopShieldingWorkerLeaseService.java:40)
        at org.gradle.api.internal.project.DefaultProjectStateRegistry$ProjectStateImpl.withProjectLock(DefaultProjectStateRegistry.java:238)
        at org.gradle.api.internal.project.DefaultProjectStateRegistry$ProjectStateImpl.withMutableState(DefaultProjectStateRegistry.java:232)
        at org.gradle.api.internal.project.DefaultProjectStateRegistry$ProjectStateImpl.withMutableState(DefaultProjectStateRegistry.java:193)
        at org.gradle.configuration.project.LifecycleProjectEvaluator$EvaluateProject.run(LifecycleProjectEvaluator.java:96)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.configuration.project.LifecycleProjectEvaluator.evaluate(LifecycleProjectEvaluator.java:68)
        at org.gradle.api.internal.project.DefaultProject.evaluate(DefaultProject.java:699)
        at org.gradle.api.internal.project.DefaultProject.evaluate(DefaultProject.java:142)
        at org.gradle.execution.TaskPathProjectEvaluator.configure(TaskPathProjectEvaluator.java:36)
        at org.gradle.execution.TaskPathProjectEvaluator.configureHierarchy(TaskPathProjectEvaluator.java:64)
        at org.gradle.configuration.DefaultProjectsPreparer.prepareProjects(DefaultProjectsPreparer.java:61)
        at org.gradle.configuration.BuildOperatingFiringProjectsPreparer$ConfigureBuild.run(BuildOperatingFiringProjectsPreparer.java:52)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.configuration.BuildOperatingFiringProjectsPreparer.prepareProjects(BuildOperatingFiringProjectsPreparer.java:40)
        at org.gradle.initialization.DefaultGradleLauncher.prepareProjects(DefaultGradleLauncher.java:204)
        at org.gradle.initialization.DefaultGradleLauncher.doClassicBuildStages(DefaultGradleLauncher.java:142)
        at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:130)
        at org.gradle.initialization.DefaultGradleLauncher.executeTasks(DefaultGradleLauncher.java:110)
        at org.gradle.internal.invocation.GradleBuildController$1.execute(GradleBuildController.java:60)
        at org.gradle.internal.invocation.GradleBuildController$1.execute(GradleBuildController.java:57)
        at org.gradle.internal.invocation.GradleBuildController$3.create(GradleBuildController.java:85)
        at org.gradle.internal.invocation.GradleBuildController$3.create(GradleBuildController.java:78)
        at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:189)
        at org.gradle.internal.work.StopShieldingWorkerLeaseService.withLocks(StopShieldingWorkerLeaseService.java:40)
        at org.gradle.internal.invocation.GradleBuildController.doBuild(GradleBuildController.java:78)
        at org.gradle.internal.invocation.GradleBuildController.run(GradleBuildController.java:57)
        at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:31)
        at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
        at org.gradle.launcher.exec.BuildOutcomeReportingBuildActionRunner.run(BuildOutcomeReportingBuildActionRunner.java:63)
        at org.gradle.tooling.internal.provider.ValidatingBuildActionRunner.run(ValidatingBuildActionRunner.java:32)
        at org.gradle.launcher.exec.BuildCompletionNotifyingBuildActionRunner.run(BuildCompletionNotifyingBuildActionRunner.java:39)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner$3.call(RunAsBuildOperationBuildActionRunner.java:51)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner$3.call(RunAsBuildOperationBuildActionRunner.java:45)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner.run(RunAsBuildOperationBuildActionRunner.java:45)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter$1.transform(InProcessBuildActionExecuter.java:50)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter$1.transform(InProcessBuildActionExecuter.java:47)
        at org.gradle.composite.internal.DefaultRootBuildState.run(DefaultRootBuildState.java:78)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:47)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:31)
        at org.gradle.launcher.exec.BuildTreeScopeBuildActionExecuter.execute(BuildTreeScopeBuildActionExecuter.java:42)
        at org.gradle.launcher.exec.BuildTreeScopeBuildActionExecuter.execute(BuildTreeScopeBuildActionExecuter.java:28)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:78)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:52)
        at org.gradle.tooling.internal.provider.SubscribableBuildActionExecuter.execute(SubscribableBuildActionExecuter.java:59)
        at org.gradle.tooling.internal.provider.SubscribableBuildActionExecuter.execute(SubscribableBuildActionExecuter.java:36)
        at org.gradle.tooling.internal.provider.SessionScopeBuildActionExecuter.execute(SessionScopeBuildActionExecuter.java:68)
        at org.gradle.tooling.internal.provider.SessionScopeBuildActionExecuter.execute(SessionScopeBuildActionExecuter.java:38)
        at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:37)
        at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:26)
        at org.gradle.tooling.internal.provider.ParallelismConfigurationBuildActionExecuter.execute(ParallelismConfigurationBuildActionExecuter.java:43)
        at org.gradle.tooling.internal.provider.ParallelismConfigurationBuildActionExecuter.execute(ParallelismConfigurationBuildActionExecuter.java:29)
        at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:60)
        at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:32)
        at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:55)
        at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:41)
        at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:48)
        at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:32)
        at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:68)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:39)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:27)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:35)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.create(ForwardClientInput.java:78)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.create(ForwardClientInput.java:75)
        at org.gradle.util.Swapper.swap(Swapper.java:38)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:75)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:55)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:63)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:82)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:52)
        at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:297)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
Caused by: org.gradle.api.internal.plugins.PluginApplicationException: Failed to apply plugin [class 'eu.xenit.gradle.alfrescosdk.AmpBasePlugin']
        at org.gradle.api.internal.plugins.DefaultPluginManager.doApply(DefaultPluginManager.java:167)
        at org.gradle.api.internal.plugins.DefaultPluginManager.apply(DefaultPluginManager.java:141)
        at eu.xenit.gradle.alfrescosdk.AmpPlugin.apply(AmpPlugin.java:31)
        at eu.xenit.gradle.alfrescosdk.AmpPlugin.apply(AmpPlugin.java:16)
        at org.gradle.api.internal.plugins.ImperativeOnlyPluginTarget.applyImperative(ImperativeOnlyPluginTarget.java:43)
        at org.gradle.api.internal.plugins.RuleBasedPluginTarget.applyImperative(RuleBasedPluginTarget.java:51)
        at org.gradle.api.internal.plugins.DefaultPluginManager.addPlugin(DefaultPluginManager.java:181)
        at org.gradle.api.internal.plugins.DefaultPluginManager.access$300(DefaultPluginManager.java:51)
        at org.gradle.api.internal.plugins.DefaultPluginManager$AddPluginBuildOperation.run(DefaultPluginManager.java:276)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.plugins.DefaultPluginManager$2.execute(DefaultPluginManager.java:159)
        at org.gradle.api.internal.plugins.DefaultPluginManager$2.execute(DefaultPluginManager.java:156)
        at org.gradle.configuration.internal.DefaultUserCodeApplicationContext.apply(DefaultUserCodeApplicationContext.java:49)
        at org.gradle.api.internal.plugins.DefaultPluginManager.doApply(DefaultPluginManager.java:156)
        at org.gradle.api.internal.plugins.DefaultPluginManager.apply(DefaultPluginManager.java:136)
        at org.gradle.plugin.use.internal.DefaultPluginRequestApplicator$3.run(DefaultPluginRequestApplicator.java:155)
        at org.gradle.plugin.use.internal.DefaultPluginRequestApplicator.applyPlugin(DefaultPluginRequestApplicator.java:231)
        ... 128 more
Caused by: org.gradle.api.InvalidUserCodeException: Please use the ObjectFactory.mapProperty() method to create a property of type Map<K, V>.
        at org.gradle.api.internal.model.DefaultObjectFactory.property(DefaultObjectFactory.java:139)
        at eu.xenit.gradle.alfrescosdk.internal.tasks.DefaultAmpSourceSetConfiguration.<init>(DefaultAmpSourceSetConfiguration.java:32)
        at eu.xenit.gradle.alfrescosdk.internal.tasks.DefaultAmpSourceSet.<init>(DefaultAmpSourceSet.java:21)
        at eu.xenit.gradle.alfrescosdk.AmpBasePlugin.lambda$configureSourceSetDefaults$2(AmpBasePlugin.java:63)
        at org.gradle.api.internal.DefaultCollectionCallbackActionDecorator$BuildOperationEmittingAction$1$1.run(DefaultCollectionCallbackActionDecorator.java:100)
        at org.gradle.configuration.internal.DefaultUserCodeApplicationContext.reapply(DefaultUserCodeApplicationContext.java:60)
        at org.gradle.api.internal.DefaultCollectionCallbackActionDecorator$BuildOperationEmittingAction$1.run(DefaultCollectionCallbackActionDecorator.java:97)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.api.internal.DefaultCollectionCallbackActionDecorator$BuildOperationEmittingAction.execute(DefaultCollectionCallbackActionDecorator.java:94)
        at org.gradle.api.internal.DefaultDomainObjectCollection.all(DefaultDomainObjectCollection.java:163)
        at eu.xenit.gradle.alfrescosdk.AmpBasePlugin.configureSourceSetDefaults(AmpBasePlugin.java:62)
        at eu.xenit.gradle.alfrescosdk.AmpBasePlugin.apply(AmpBasePlugin.java:50)
        at eu.xenit.gradle.alfrescosdk.AmpBasePlugin.apply(AmpBasePlugin.java:32)
        at org.gradle.api.internal.plugins.ImperativeOnlyPluginTarget.applyImperative(ImperativeOnlyPluginTarget.java:43)
        at org.gradle.api.internal.plugins.RuleBasedPluginTarget.applyImperative(RuleBasedPluginTarget.java:51)
        at org.gradle.api.internal.plugins.DefaultPluginManager.addPlugin(DefaultPluginManager.java:181)
        at org.gradle.api.internal.plugins.DefaultPluginManager.access$300(DefaultPluginManager.java:51)
        at org.gradle.api.internal.plugins.DefaultPluginManager$AddPluginBuildOperation.run(DefaultPluginManager.java:276)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.plugins.DefaultPluginManager$2.execute(DefaultPluginManager.java:159)
        at org.gradle.api.internal.plugins.DefaultPluginManager$2.execute(DefaultPluginManager.java:156)
        at org.gradle.configuration.internal.DefaultUserCodeApplicationContext.apply(DefaultUserCodeApplicationContext.java:49)
        at org.gradle.api.internal.plugins.DefaultPluginManager.doApply(DefaultPluginManager.java:156)
        ... 150 more

```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,12 +1,7 @@
 language: java
 
-before_cache:
-  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
-  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
-
 cache:
   directories:
-    - $HOME/.gradle/caches/
     - $HOME/.gradle/wrapper/
 
 script: ./gradlew check -i

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,11 @@
 # Alfresco Gradle SDK - Changelog
 
+## Unreleased
+
+### Fixed
+
+ * [#33](https://github.com/xenit-eu/alfresco-gradle-sdk/pull/33) Support Gradle 6.0
+
 ## Version 1.0.0 - 2019-09-19
 
 ### Fixed

--- a/src/integrationTest/java/eu/xenit/gradle/alfrescosdk/AbstractIntegrationTest.java
+++ b/src/integrationTest/java/eu/xenit/gradle/alfrescosdk/AbstractIntegrationTest.java
@@ -28,7 +28,8 @@ public abstract class AbstractIntegrationTest {
     @Parameters(name = "Gradle v{0}")
     public static Collection<Object[]> testData() {
         return Arrays.asList(new Object[][]{
-                {"5.6.2"},
+                {"6.0"},
+                {"5.6.4"},
                 {"5.5.1"},
                 {"5.4.1"},
                 {"5.3.1"},

--- a/src/integrationTest/java/eu/xenit/gradle/alfrescosdk/AbstractIntegrationTest.java
+++ b/src/integrationTest/java/eu/xenit/gradle/alfrescosdk/AbstractIntegrationTest.java
@@ -17,6 +17,7 @@ import org.junit.runners.Parameterized.Parameters;
 
 @RunWith(Parameterized.class)
 public abstract class AbstractIntegrationTest {
+
     @Rule
     public TemporaryFolder testFolder = new TemporaryFolder();
 
@@ -28,7 +29,7 @@ public abstract class AbstractIntegrationTest {
     @Parameters(name = "Gradle v{0}")
     public static Collection<Object[]> testData() {
         return Arrays.asList(new Object[][]{
-                {"6.0"},
+                {"6.0.1"},
                 {"5.6.4"},
                 {"5.5.1"},
                 {"5.4.1"},

--- a/src/integrationTest/java/eu/xenit/gradle/alfrescosdk/Examples.java
+++ b/src/integrationTest/java/eu/xenit/gradle/alfrescosdk/Examples.java
@@ -14,6 +14,8 @@ import java.nio.file.Paths;
 import java.util.Properties;
 import java.util.function.Predicate;
 import org.gradle.util.GUtil;
+import org.gradle.util.GradleVersion;
+import org.junit.Assume;
 import org.junit.Test;
 
 public class Examples extends AbstractIntegrationTest {
@@ -52,6 +54,9 @@ public class Examples extends AbstractIntegrationTest {
 
     @Test
     public void testLegacyDeProject() throws IOException {
+        // Gradle 5.1 has problems with memory usage with the bnd plugin
+        Assume.assumeTrue("Gradle version >= 5.2",
+                GradleVersion.version(gradleVersion).compareTo(GradleVersion.version("5.2")) >= 0);
         testProjectFolder(EXAMPLES.resolve("legacy-de-project"));
 
         Path buildFolder = projectFolder.toPath().resolve("build");
@@ -77,6 +82,9 @@ public class Examples extends AbstractIntegrationTest {
 
     @Test
     public void testSimpleDeProject() throws IOException {
+        // Gradle 5.1 has problems with memory usage with the bnd plugin
+        Assume.assumeTrue("Gradle version >= 5.2",
+                GradleVersion.version(gradleVersion).compareTo(GradleVersion.version("5.2")) >= 0);
         testProjectFolder(EXAMPLES.resolve("simple-de-project"));
 
         Path buildFolder = projectFolder.toPath().resolve("build");

--- a/src/integrationTest/resources/examples/legacy-de-project/build.gradle
+++ b/src/integrationTest/resources/examples/legacy-de-project/build.gradle
@@ -1,18 +1,8 @@
-buildscript {
-    repositories {
-        mavenCentral()
-    }
-    dependencies {
-        classpath "eu.xenit.de:gradle-plugin:1.7.6"
-    }
-}
-
 plugins {
+    id 'eu.xenit.de' version '2.0.1'
     id 'eu.xenit.alfresco'
     id 'eu.xenit.amp'
 }
-
-apply plugin: 'alfresco-dynamic-extension'
 
 ampConfig {
     dynamicExtension = true

--- a/src/integrationTest/resources/examples/simple-de-project/build.gradle
+++ b/src/integrationTest/resources/examples/simple-de-project/build.gradle
@@ -1,13 +1,5 @@
-buildscript {
-    repositories {
-        mavenCentral()
-    }
-    dependencies {
-        classpath "eu.xenit.de:gradle-plugin:1.7.6"
-    }
-}
-
 plugins {
+    id 'eu.xenit.de' version '2.0.1'
     id 'eu.xenit.alfresco'
     id 'eu.xenit.amp'
 }

--- a/src/main/java/eu/xenit/gradle/alfrescosdk/AmpBasePlugin.java
+++ b/src/main/java/eu/xenit/gradle/alfrescosdk/AmpBasePlugin.java
@@ -115,10 +115,10 @@ public class AmpBasePlugin implements Plugin<Project> {
     }
 
     private Provider<WriteProperties> createWritePropertiesTask(String taskName, String sourceSetName, String fileName,
-            Provider<Properties> properties) {
+            Provider<Map<String, String>> properties) {
         return project.getTasks().register(taskName, WriteProperties.class, writeProperties -> {
             writeProperties.setDescription("Creates " + fileName + " for " + sourceSetName);
-            writeProperties.setProperties(properties.map(AmpBasePlugin::propertiesToMap).get());
+            writeProperties.setProperties((Map<String, Object>)(Map)properties.get());
             writeProperties.setOutputFile(project.getBuildDir().toPath().resolve(taskName).resolve(fileName).toFile());
         });
     }

--- a/src/main/java/eu/xenit/gradle/alfrescosdk/internal/tasks/DefaultAmpSourceSetConfiguration.java
+++ b/src/main/java/eu/xenit/gradle/alfrescosdk/internal/tasks/DefaultAmpSourceSetConfiguration.java
@@ -2,10 +2,13 @@ package eu.xenit.gradle.alfrescosdk.internal.tasks;
 
 import eu.xenit.gradle.alfrescosdk.tasks.AmpSourceSetConfiguration;
 import java.io.File;
+import java.util.Collections;
+import java.util.Map;
 import java.util.Properties;
 import org.gradle.api.Action;
 import org.gradle.api.Project;
 import org.gradle.api.file.SourceDirectorySet;
+import org.gradle.api.provider.MapProperty;
 import org.gradle.api.provider.Property;
 import org.gradle.api.provider.Provider;
 import org.gradle.util.GUtil;
@@ -15,8 +18,8 @@ public class DefaultAmpSourceSetConfiguration implements AmpSourceSetConfigurati
     private final SourceDirectorySet config;
     private final SourceDirectorySet web;
     private final Project project;
-    private final Property<Properties> moduleProperties;
-    private final Property<Properties> fileMappingProperties;
+    private final MapProperty<String, String> moduleProperties;
+    private final MapProperty<String, String> fileMappingProperties;
     private final Property<Boolean> dynamicExtension;
 
     public DefaultAmpSourceSetConfiguration(Project project) {
@@ -29,12 +32,10 @@ public class DefaultAmpSourceSetConfiguration implements AmpSourceSetConfigurati
         web = project.getObjects().sourceDirectorySet("web", "Alfresco AMP web");
 
         //alfresco module.properties file
-        moduleProperties = project.getObjects().property(Properties.class);
-        moduleProperties.set(new Properties());
+        moduleProperties = project.getObjects().mapProperty(String.class, String.class).empty();
 
         //file-mapping properties
-        fileMappingProperties = project.getObjects().property(Properties.class);
-        fileMappingProperties.set(new Properties());
+        fileMappingProperties = project.getObjects().mapProperty(String.class, String.class).empty();
 
         //dynamic extension related options.
         dynamicExtension = project.getObjects().property(Boolean.class);
@@ -49,8 +50,9 @@ public class DefaultAmpSourceSetConfiguration implements AmpSourceSetConfigurati
 
     @Override
     public AmpSourceSetConfiguration module(File moduleProperties) {
-        this.moduleProperties.set(project.provider(() -> GUtil.loadProperties(moduleProperties)));
-        return this;
+        return module(properties -> {
+            properties.putAll(GUtil.loadProperties(moduleProperties));
+        });
     }
 
     @Override
@@ -58,12 +60,12 @@ public class DefaultAmpSourceSetConfiguration implements AmpSourceSetConfigurati
         moduleProperties.set(project.provider(() -> {
             Properties newProperties = new Properties();
             configure.execute(newProperties);
-            return newProperties;
+            return Collections.checkedMap((Map<String, String>) (Map) newProperties, String.class, String.class);
         }));
         return this;
     }
 
-    public Provider<Properties> getModuleProperties() {
+    public MapProperty<String, String> getModuleProperties() {
         return moduleProperties;
     }
 
@@ -75,8 +77,9 @@ public class DefaultAmpSourceSetConfiguration implements AmpSourceSetConfigurati
 
     @Override
     public AmpSourceSetConfiguration fileMapping(File fileMappingProperties) {
-        this.fileMappingProperties.set(project.provider(() -> GUtil.loadProperties(fileMappingProperties)));
-        return this;
+        return fileMapping(properties -> {
+            properties.putAll(GUtil.loadProperties(fileMappingProperties));
+        });
     }
 
     @Override
@@ -84,13 +87,13 @@ public class DefaultAmpSourceSetConfiguration implements AmpSourceSetConfigurati
         fileMappingProperties.set(project.provider(() -> {
             Properties newProperties = new Properties();
             configure.execute(newProperties);
-            return newProperties;
+            return Collections.checkedMap((Map<String, String>) (Map) newProperties, String.class, String.class);
         }));
         return this;
     }
 
 
-    public Provider<Properties> getFileMappingProperties() {
+    public MapProperty<String, String> getFileMappingProperties() {
         return fileMappingProperties;
     }
 

--- a/src/main/java/eu/xenit/gradle/alfrescosdk/tasks/Amp.java
+++ b/src/main/java/eu/xenit/gradle/alfrescosdk/tasks/Amp.java
@@ -40,9 +40,9 @@ public class Amp extends Zip {
     @Deprecated
     private Supplier<File> config;
 
-    private RegularFileProperty moduleProperties = getProject().getLayout().fileProperty();
+    private RegularFileProperty moduleProperties = getProject().getObjects().fileProperty();
 
-    private RegularFileProperty fileMappingProperties = getProject().getLayout().fileProperty();
+    private RegularFileProperty fileMappingProperties = getProject().getObjects().fileProperty();
 
     private ConfigurableFileCollection deBundles = getProject().files();
 

